plugins/woocommerce/includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -1866,6 +1866,13 @@ protected function get_wp_query_args( $query_vars ) {
 				'field'    => 'slug',
 				'terms'    => $query_vars['category'],
 			);
+		} elseif ( ! empty( $query_vars['category_id'] ) ) {
+			$wp_query_args['tax_query'][] = array(
+				'taxonomy' => 'product_cat',
+				'field'    => 'term_id',
+				'terms'    => $query_vars['category_id'],
+			);
+			unset( $wp_query_args[ 'category_id' ] ) ;
```

Comment:
Unfortunately, I can't remember _why_ that was originally included. Would probably need to test with and without it at this point.

Reply:
Removing the `unset` for `category_id` has no effect on the now-included unit tests, while removing the `unset` for `tag_id` makes the corresponding test fail. I wonder if there's something going on in the inner working of WordPress where that `tag_id` overrides something that Woo is doing? 🤔

Reply:
> I wonder if there's something going on in the inner working of WordPress where that tag_id overrides something that Woo is doing? 🤔

It looks like that is [indeed the case](https://github.com/WordPress/WordPress/blob/1050b47e063107e61feccfbe2e58e47c3120410e/wp-includes/class-wp-query.php#L1319-L1325).

🤔 I wonder if it would be better to avoid potential collisions altogether by changing the new query var names here to `product_category_id` and `product_tag_id`... what do you think?

Reply:
@coreymckrill great find, thank you! I agree we should aim to not interfere with that. I've renamed both as you suggested. 👍